changelog/fix-8179-remove-deposits-api

```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: dev
+
+Remove unused `deposits/overview` API
```

Comment:
This is not super clear, though maybe we don't need a changelog.

The change here is that we are removing this unused API and dead code related to it in various layers – JS/front end, plugin client, and server. Changelogs for each part of that are confusing IMO!

Reply:
I think we do need a changelog. 

#8178 was removing dead front-end code that isn't exposed anywhere. This PR removes an API that 3rd parties could be using.

What is unclear about it? Is it the wording?

Reply:
> What is unclear about it? Is it the wording?

Just could be more specific – it assumes the reader understands the whole thing that is getting removed (of which this is a part).

> This PR removes an API that 3rd parties could be using.

Write with those potential clients in mind – what do they depend on that is changing? Be specific about the kind of API – is this an HTTP / REST API exposed on the merchant site? Or a mirror of a WooPayments server API? (Or a hook/WP API – api can have many meanings?) If it's potentially used, we should be clearer (and probably not focus on the fact that _we_ aren't using the API).